docs/gaps/versioning-policy/README.md

```diff
@@ -0,0 +1,215 @@
+required developer, the root key holders may add their signature to all
+metadata, rule files included, for the first policy state with the newer major
+version.
+
+More generally, this updates the policy state verification behavior to allow
+signatures on any policy metadata by the root of key holders, even if they're
+not part of the direct delegation on that policy metadata. Note that a threshold
+of root key holders are required in such instances.
+
+#### gittuf Implementation Versions
+
+The gittuf implementation or client version conveys the behavior of gittuf
+workflows it invokes. In that sense, this encompasses the version of the
+predicate and policy metadata, as the client selects the version of metadata it
+signs and consumes.
+
+Significantly, the verification behavior of gittuf is only versioned via the
+client's version, as this is not encoded as metadata itself. If the verification
+behavior changes for existing policy and attestation predicate versions in a way
+that a change that was previously allowed is now disallowed or vice versa, the
+implementation's major version must be bumped. The gittuf client may check for
+the existance of a newer version and inform the user. The client must not
+attempt to update itself.
+
+## Motivation
+
+Different clients (on different versions) may default to different versions of a
+predicate for the same context. For example, Alice may use an older version of
+the reference authorization predicate to approve a change. Bob, who wants to
+approve the same change, may have a newer version of the gittuf client which
+uses a newer version of the predicate. Similarly, Alice may want to amend a
+policy file that was last modified by Bob, and thus is of a newer version
+unsupported by Alice's client. Handling these types of version mismatches is
+imperative to ensure a developer using an older client does not cause
+downgrade-style attacks, where the repository's metadata is rolled back to an
+older version with fewer security guarantees.
+
+## Reasoning
+
+This GAP introduces ideas that balance security and usability. Being very strict
+with policy version changes can lead to constant incompatibilities and break
+developers' workflows.
+
+## Backwards Compatibility
+
+This GAP does not have impact on backwards compatibility of the existing gittuf
+design and implementation. Indeed, the GAP introduces granular versioning of
+metadata and behavior, which impacts how gittuf handles backwards compatibility
+going forward.
+
+## Security
+
+The GAP considers the impact of mixed versioning for attestation predicates and
+the policy metadata. Generally, the GAP proposes mechanisms to prevent rollback
+attacks for both types of metadata. In addition, the GAP ensures clients fail in
+actions that modify or verify metadata the client does not understand. This
+ensures that an older client does not allow a change that would be prevented
+with newer security features the client does not understand.
+
```

Comment:
Would it be sane for a future version of gittuf to also be able to set a minimum metadata version to deprecate them?  For example, v4.0 clients no longer need to support v0.9 metadata.  This would make it so clients could remove stale policy verification logic.

Reply:
I think no. We would of course disallow creating new policy metadata using v0.9 instead of vLatest supported by v4.0 client, but we should support it so that a repository which was created when v0.9 was latest can continue to have that old policy metadata still verified.

Reply:
I'm about to push a rework that makes this explicit, in that clients must continue supporting older versions to ensure verification of historic activity works.